Core/HLE/sceKernelModule.cpp

```diff
@@ -1806,14 +1799,11 @@ bool __KernelLoadExec(const char *filename, u32 paramPtr, std::string *error_str
 		return false;
 	}
 
-	u32 handle = pspFileSystem.OpenFile(filename, FILEACCESS_READ);
-
-	u8 *temp = new u8[(int)info.size + 0x01000000];
-
-	pspFileSystem.ReadFile(handle, temp, (size_t)info.size);
-
 	PSP_SetLoading("Loading modules...");
-	PSPModule *module = __KernelLoadModule(temp, (size_t)info.size, 0, error_string);
+	size_t size = fileData.size();
+	// TODO: Why do we add this padding?  Crash avoidance?
+	fileData.resize(fileData.size() + 0x01000000);
```

Comment:
This looks dubious and a bit excessive indeed. Would prefer to remove...

Reply:
Yeah, it's 16 MB.  Been there for a long time it seems.  I guess I can just remove, I'd kinda rather do that separately in case something I did here breaks save data or causes crashes.

-[Unknown]

Reply:
Agreed.